src/laser_geometry.cpp

```diff
@@ -43,10 +43,20 @@ namespace laser_geometry
 
     // Fill the ranges matrix
     for (unsigned int index = 0; index < scan_in.ranges.size(); index++)
+    {
+      if (preservative)
+      {
+        ranges (0, index) = (double) (scan_in.ranges[index] == std::numeric_limits<float>::infinity()
+                          ? scan_in.range_max : scan_in.ranges[index]);
+        ranges (1, index) = (double) (scan_in.ranges[index] == std::numeric_limits<float>::infinity()
```

Comment:
Why do you do this when `preservative == true`? I mean, it looks like you're doing two different changes here:

- Expose `preservative`
- Map `+Inf`, `-Inf`, `NaN` to finite values, for driver following REP-117 (http://www.ros.org/reps/rep-0117.html).

For the REP-117, when `preservative == true`, the user might still want to have the non-finite values on the result.